nncf/openvino/quantization/quantize_model_if_op.py

```diff
@@ -0,0 +1,226 @@
+from itertools import islice
+from pathlib import Path
+from typing import Callable, Dict, List, Optional, Set, Tuple, TypeVar
+
+import openvino.runtime as ov
+from tqdm import tqdm
+
+from nncf import Dataset
+from nncf.common import factory
+from nncf.common.engine import Engine
+from nncf.common.factory import NNCFGraphFactory
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.model_transformer import ModelTransformer
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.common.logging import nncf_logger
+from nncf.common.tensor_statistics.statistic_point import StatisticPointsContainer
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVIfMetatype
+from nncf.openvino.graph.node_utils import has_if_op
+from nncf.openvino.graph.transformations.commands import OVExtractIfSubgraphCommand
+from nncf.openvino.graph.transformations.commands import OVOutputInsertionCommand
+from nncf.openvino.graph.transformations.commands import OVTargetPoint
+from nncf.openvino.graph.transformations.commands import OVUpdateIfSubgraphCommand
+from nncf.quantization.algorithms.algorithm import Algorithm
+
+
+def _make_dataset_for_children_model(
+    engine: Engine,
+    calibration_dataset: Dataset,
+    if_cond_input_name: str,
+    then_model_input_names,
+    else_model_input_names,
+    subset_size: int,
+) -> Tuple[Dataset, Dataset]:
+    """
+    Returns dataset for a child model.
+
+    :param engine: Engine to infer parent model to obtain dataitems for a child dataset.
+    :param calibration_dataset: Dataset to infer parent model.
+    :param if_cond_input_name: Input name of If node condition.
+    :param child_model_input_names: - Names of inputs for child model
+    (should be in the order of passing them to a model).
+    :param if_submodel_condition: If node submodel condition.
+    :return Dataset: Dataset for child model.
+    """
+    then_dataset, else_dataset = [], []
+    calibration_dataset_size = (
+        min(subset_size, calibration_dataset.get_length())
+        if calibration_dataset.get_length() is not None
+        else subset_size
+    )
+    for input_data in tqdm(
```

Comment:
Minor. Shouldn't we update it with the `rich` progress bar?

Reply:
Done